### PR TITLE
Documentation: Authentication and Authorization improvements

### PR DESCRIPTION
**Pull request by charlietfe**
## Description
This PR integrates **improvements on the `Authentication and Authorization` section,** clarifying some of the aspects of the feature, reorganising the text to be easier to understand and fixing some mistakes.

## Changes
- Restructured `Authentication and authorization` section.
- Improved `Custom authentication` section, explaining a little bit better some of the aspects of the feature and fixing some misleading explanations or orthographic mistakes.

## Checks
- [X] Project Builds
- [X] Project passes tests and checks
- [X] Updated documentation accordingly

### CODE REVIEW

File: docs/chapters/04_features.md [hunk 1]
```diff
@@ -134,20 +138,20 @@ export class UpdateUser {
 }
 ```
 
-Your token should include a property `custom:role` with the value `Admin` or `User`. Here is an example of a Firebase token:
+Your token should include a property specified in `rolesClaim` with the value `Admin` or `User`. Here is an example of a Firebase token:
 
 ```json
 {
-  "custom:role": "User",
```
Comment (charlietfe) [inline docs/chapters/04_features.md:141]: Finally we've fixed this syntax issue.

**Review by AdrianLorenzoDev — APPROVED** [review]
LGTM, just one typo! :D

**Review by davidverdu — APPROVED** [review]
LGTM

File: docs/chapters/04_features.md [hunk 2]
```diff
@@ -83,28 +84,30 @@ Users that sign up with their emails will receive a confirmation link in their i
...
 
-You can use the **JWT authorization mode** to authorize all incoming Booster requests. Your auth server will return JWT tokens
-wich will be decoded internally by Booster, after that, the required roles will be matched with the contained claims
-inside that token.
+Booster provides a **JWT authorization mode** to authorize all incoming Booster requests using the server you decide. Your authentication server will provide JWT tokens that you can use with Booster. Your application will decode your token and verify its validity with your server, and then, the required roles will be matched with the claims contained in the token.
+
+In that way, you can use different authentication providers, like Auth0, Firebase, Cognito, or create your own, without the need of a specific rocket implementation.
 
-In that way, you can use different auth providers, like Auth0, Firebase, Cognito, create your own or simply use our [Authentication Rocket](https://github.com/boostercloud/rocket-auth-aws-infrastructure), which is our recommended solution that works great with Booster.
+Note:
```
Comment (Moneyba) [inline docs/chapters/04_features.md:101]: For creating a note comment, you can use  a dedicated alarm for that using `> [!NOTE]`

**Review by Moneyba — APPROVED** [review]
LGTM. I just left a small comment.

**Pull request merged**